scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/context/TaskContextVariableExtractor.java

```diff
@@ -122,11 +122,17 @@
 
     public Map<String, Serializable> extractScopeVariables(TaskContext taskContext) {
         Map<String, Serializable> variables = new HashMap<>();
-
+        Map<String, Serializable> previousVariables = null;
+        try {
+            previousVariables = extractVariables(taskContext, false);
+        } catch (IOException | ClassNotFoundException e) {
+            logger.error("Unable to extract variables", e);
+        }
         // variables from task definition
         if (taskContext.getInitializer().getTaskVariables() != null) {
```

Comment:
why if (taskContext.getInitializer().getTaskVariables() != null)
instead of
 if (taskContext.getInitializer() != null && taskContext.getInitializer().getTaskVariables() != null)